### DISCUSSION TITLE
Branch Question

### DISCUSSION THREAD

**Discussion by Crip-Zone** [community/community#55702] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I run a website that creates themes for Simple Machines Forums. In order to keep the various themes separated, do I need to create various repositories or various branches for each theme?

Do I need to create one repository and then one separate branch for each theme or have sub-branches from main branches for each theme?

Thanks in advance. :)

**Comment by airtower-luna** [comment, marked as answer]
> In order to keep the various themes separated, do I need to create various repositories or various branches for each theme?

I'd recommend separate repositories. Branches are usually used for different versions of the same thing (e.g. as you work on a new feature, or maintain an older version). Different themes are different things and should probably live in different repositories.

> Do I need to create one repository and then one separate branch for each theme or have sub-branches from main branches for each theme?

There are no sub-branches in Git. A branch is a branch. I wonder if you're confusing branches with directories? :thinking: 

If you're new to Git, I highly recommend you take a look at the [Git Book](https://git-scm.com/book/en/v2) (it's a website, but you can download an ebook version if you prefer). The first 3 chapters cover the basics. :slightly_smiling_face:

**Reply by Crip-Zone** [reply]
Thanks for the reply. Yes, I probably am confusing branches with directories, since I use them a lot. LOL

**Reply by airtower-luna** [reply]
If you need separate *directories* or repositories depends mostly on how people are gonna work with the themes, I suppose. :slightly_smiling_face:

**Reply by Crip-Zone** [reply]
Well, the themes are for SMF forums. Each theme is installed under separate directories within the main forum. Since we have quite a number of themes I was just wondering the best way to go about it.

I just need to keep versions for each theme created separated and documented while making sure they are not integrated with the other themes so the code doesn't get mixed up. LOL
